src/NServiceBus.RavenDB/Timeouts/QueryTimeouts.cs

```diff
@@ -21,93 +24,85 @@ public QueryTimeouts(IDocumentStore documentStore, string endpointName)
             TriggerCleanupEvery = TimeSpan.FromMinutes(2);
             CleanupGapFromTimeslice = TimeSpan.FromMinutes(1);
             shutdownTokenSource = new CancellationTokenSource();
+            logger = LogManager.GetLogger<QueryTimeouts>();
         }
 
         public TimeSpan CleanupGapFromTimeslice { get; set; }
         public TimeSpan TriggerCleanupEvery { get; set; }
+        public Func<DateTime> GetUtcNow { get; set; } = () => DateTime.UtcNow;
 
         public async Task<TimeoutsChunk> GetNextChunk(DateTime startSlice)
         {
-            var now = DateTime.UtcNow;
+            var now = GetUtcNow();
             List<TimeoutsChunk.Timeout> results;
 
             // Allow for occasionally cleaning up old timeouts for edge cases where timeouts have been
             // added after startSlice have been set to a later timout and we might have missed them
             // because of stale indexes.
             if (lastCleanupTime == DateTime.MinValue || lastCleanupTime.Add(TriggerCleanupEvery) < now)
             {
-                results = await GetCleanupChunk(startSlice).ConfigureAwait(false);
+                results = await GetCleanupChunk(now).ConfigureAwait(false);
```

Comment:
Nitpicking:
- Most of the time the first condition will be false, so might be better to switch the order in the OR.
- lastCleanupTime.Add(TriggerCleanupEvery) < now - I had to pause for a minute and think what's going on, I think it'll be a bit more clearer if we renamed `TriggerCleanupEvery` to something like `CleanupInterval`.